src/permissionSelectors.js

```diff
@@ -40,6 +40,63 @@ export function roleIsAtLeast(thisRole: RoleT, thresholdRole: RoleT): boolean {
   return thisRole <= thresholdRole; // Roles with more privilege have lower numbers.
 }
 
+/**
+ * Whether the self-user is authorized to create or edit a stream to be
+ *   public.
+ *
+ * Note: This isn't about web-public streams. For those, see
+ * getCanCreateWebPublicStreams.
+ */
+export function getCanCreatePublicStreams(state: PerAccountState): boolean {
+  const { createPublicStreamPolicy } = getRealm(state);
+  const role = getOwnUserRole(state);
+
+  switch (createPublicStreamPolicy) {
+    case 4: // ModeratorOrAbove
+      return roleIsAtLeast(role, Moderator);
+    case 3: // FullMemberOrAbove
+      // TODO: Implement properly.
+      return roleIsAtLeast(role, Member);
```

Comment:
I've got an implementation in my next revision. :) I was mostly concerned with client-side logic (how stale will the value be, since it depends on the current time), but I think those concerns end up being pretty small.